plutus-ir/test/recursion/even3.plc.golden

```diff
@@ -1,79 +1,609 @@
 (program 1.0.0
```

Comment:
I thought of the problem of not duplicating tuples, solved it and then stupidly forgot to implement the solution. Thanks for catching this. Updated the file, it's now shorter than it was originally, because we do not generate unnecessary instantiations.